app-creator/client/src/data/templates.ts

```diff
@@ -0,0 +1,54 @@
+import { TemplateItem, database } from '../client/fetch-templates';
+
+class TemplatesManager {
+  private templates: TemplateItem[] = [];
+
+  getTemplates() {
+    return this.templates;
+  }
+
+  async fetchTemplates(
+    forceReload: boolean = false,
+    onError?: () => void
```

Comment:
Since this is an async function, it's best to rethrow the error instead of accepting a callback, which the caller can `catch` to handle the error case. It also means the method will not return the templates as you originally designed, but I think that the extra logic there would make it more explicit about what's going on if we end up removing the local cache later.